ios/Codeforces Watcher/Application/features/news/cells/PinnedPostView.swift

```diff
@@ -0,0 +1,219 @@
+import SwiftUI
+
+struct PinnedPostView: View {
+    
+    var post: NewsItem.PinnedItem
+    var onNews: (_ title: String, _ link: String) -> Void = { _, _ in }
+    
+    @State private var isHidden = true
+    
+    private let animationDuration = 0.3
+    
+    var body: some View {
+        if isHidden {
+            Button(action: {
+                toggle()
+            }, label: {
+                HiddenPinnedPostView(title: post.title)
+            })
+        } else {
+            VisiblePinnedPostView(
+                post: post,
+                onNews: onNews,
+                onCrossButton: {
+                    toggle()
+                }
+            )
+        }
+    }
+    
+    private func toggle() {
+        withAnimation(.easeIn(duration: animationDuration)) {
+            isHidden.toggle()
+        }
+    }
+}
+
+fileprivate struct HiddenPinnedPostView: View {
+    
+    var title: String
+    
+    var body: some View {
+        HStack(alignment: .top, spacing: 12) {
+            LogoView()
+            
+            TitleView
+        }
+        .padding(.horizontal, 12)
+        .frame(height: 60)
+        .background(BackgroundView)
+        .cornerRadius(20)
+        .environment(\.colorScheme, .light)
+    }
+    
+    @ViewBuilder
+    private var TitleView: some View {
+        VStack(alignment: .leading, spacing: 0) {
+            LinearGradient(
+                gradient: Gradient(colors: [
+                    Palette.redGradient.swiftUIColor,
+                    Palette.yellowGradient.swiftUIColor,
+                    Palette.blueGradient.swiftUIColor
+                ]),
+                startPoint: .leading,
+                endPoint: .trailing
+            ).mask(
+                CommonText(title)
+                    .font(.midHeaderSemibold2)
+                    .frame(maxWidth: .infinity, alignment: .leading)
+            )
+                .frame(height: 18)
```

Comment:
IMO, having `.frame` on the new line and padded by more spaces make us think that it's related to `CommonText` and not to actual `LinearGradient`. WDYT about having `).frame` instead?

Reply:
You are right🤔